go.mod

```diff
@@ -0,0 +1,70 @@
+	github.com/dustin/go-humanize v0.0.0-20180421182945-02af3965c54e
+	github.com/eknkc/amber v0.0.0-20171010120322-cdade1c07385 // indirect
+	github.com/etcd-io/gofail v0.0.0-20180808172546-51ce9a71510a
+	github.com/ghodss/yaml v1.0.0
+	github.com/gogo/protobuf v1.0.0
+	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b // indirect
+	github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4 // indirect
+	github.com/golang/protobuf v1.2.0
+	github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db // indirect
+	github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c
+	github.com/gorilla/context v0.0.0-20160226214623-1ea25387ff6f // indirect
+	github.com/gorilla/mux v1.6.1
+	github.com/gorilla/websocket v1.2.0 // indirect
+	github.com/grpc-ecosystem/go-grpc-prometheus v0.0.0-20160910222444-6b7015e65d36
+	github.com/grpc-ecosystem/grpc-gateway v1.4.1 // indirect
+	github.com/inconshreveable/mousetrap v1.0.0 // indirect
+	github.com/jonboulle/clockwork v0.1.0 // indirect
+	github.com/mattn/go-shellwords v1.0.3
+	github.com/matttproud/golang_protobuf_extensions v1.0.0 // indirect
+	github.com/montanaflynn/stats v0.0.0-20151014174947-eeaced052adb
+	github.com/onsi/gomega v1.4.2 // indirect
+	github.com/opentracing/opentracing-go v1.0.2
+	github.com/pingcap/check v0.0.0-20171206051426-1c287c953996
+	github.com/pingcap/errcode v0.0.0-20180921232412-a1a7271709d9
+	github.com/pingcap/errors v0.10.1 // indirect
+	github.com/pingcap/kvproto v0.0.0-20181010074705-0ba3ca8a6e37
+	github.com/pkg/errors v0.8.0
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/prometheus/client_golang v0.8.0
+	github.com/prometheus/client_model v0.0.0-20171117100541-99fa1f4be8e5 // indirect
+	github.com/prometheus/common v0.0.0-20180426121432-d811d2e9bf89 // indirect
+	github.com/prometheus/procfs v0.0.0-20180408092902-8b1c2da0d56d // indirect
+	github.com/sirupsen/logrus v1.0.5
+	github.com/soheilhy/cmux v0.1.4 // indirect
+	github.com/spf13/cobra v0.0.2
+	github.com/spf13/pflag v1.0.1
+	github.com/stretchr/testify v1.2.2 // indirect
+	github.com/syndtr/goleveldb v0.0.0-20180815032940-ae2bd5eed72d
+	github.com/tmc/grpc-websocket-proxy v0.0.0-20171017195756-830351dc03c6 // indirect
+	github.com/ugorji/go v1.1.1 // indirect
+	github.com/unrolled/render v0.0.0-20171102162132-65450fb6b2d3
+	github.com/urfave/negroni v0.3.0
+	github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18 // indirect
+	go.uber.org/atomic v1.3.2 // indirect
+	go.uber.org/multierr v1.1.0 // indirect
+	go.uber.org/zap v1.8.0
+	golang.org/x/crypto v0.0.0-20180503215945-1f94bef427e3 // indirect
+	golang.org/x/net v0.0.0-20180906233101-161cd47e91fd
+	golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2 // indirect
+	google.golang.org/genproto v0.0.0-20180427144745-86e600f69ee4 // indirect
+	google.golang.org/grpc v1.12.2
+	gopkg.in/airbrake/gobrake.v2 v2.0.9 // indirect
+	gopkg.in/gemnasium/logrus-airbrake-hook.v2 v2.1.2 // indirect
```

Comment:
How did these airbrakes end up getting added? They weren't there before and shouldn't be PD dependencies.

Reply:
From `go mod why gopkg.in/gemnasium/logrus-airbrake-hook.v2` we can see it is imported by `github.com/sirupsen/logrus.test`.

Reply:
I did run that, but I didn't know what logrus.test is. It isn't listed in our dependencies.

It seems that go modules adds all transitive testing dependencies.
* https://www.reddit.com/r/golang/comments/98hbvk/go_modules_how_to_deal_with_test_dependencies/
* https://github.com/golang/go/issues/26913
* https://github.com/golang/go/issues/26626

This makes me prefer dep to go modules for now until they get more feedback on go modules.

Reply:
I looked at the logrus repo and I don't see how it can reference airbrake (testing or not): https://github.com/sirupsen/logrus/search?q=airbrake&unscoped_q=airbrake

This actually seems like an issue with how they [defined their go.mod](https://github.com/sirupsen/logrus/pull/817/files). Perhaps you can ask them about it?

I still don't understand why go mod tidy doesn't remove this since it isn't needed.

```
GO111MODULE=on go mod why github.com/sirupsen/logrus.tesi
go: finding github.com/golang/protobuf/proto/testdata latest
go: finding github.com/golang/protobuf/proto latest
# github.com/sirupsen/logrus.tesi
(main module does not need package github.com/sirupsen/logrus.tesi)
```

Reply:
It looks like there is actually an *open* issue about testing dependencies here: https://github.com/golang/go/issues/26955. It is milestoned for 1.12 although not necessarily accepted.

Reply:
It looks the best effort now is to keep the `vendor/` directory, setup `-mod=vendor` in Makefile, and only use `go mod` when we need to update imports. The update procedure should include `./hack/clean-vendor` to cleanup test files.
I'll take a try.

Reply:
:+1: but I don't think clean-vendor will end up removing exteraneous deps like airbrake. And now `go mod` is built-in automatically to go commands, so I don't know if it is possible to avoid it updating imports.